pandas/tests/reshape/merge/test_merge.py

```diff
@@ -2082,3 +2082,55 @@ def test_merge_equal_cat_dtypes2():
 
     # Categorical is unordered, so don't check ordering.
     tm.assert_frame_equal(result, expected, check_categorical=False)
+
+
+def test_right_merge_preserves_row_order():
+    population = [
+        ("Jenn", "Jamaica", 3),
+        ("Beth", "Bulgaria", 7),
+        ("Carl", "Canada", 30),
+    ]
+    columns = ["name", "country", "population"]
+    pop = DataFrame.from_records(population, columns=columns)
+
+    people = [("Abe", "America"), ("Beth", "Bulgaria"), ("Carl", "Canada")]
+    columns = ["name", "country"]
+    ppl = DataFrame.from_records(people, columns=columns)
+
+    expected_data = [
+        ("Abe", "America", np.nan),
+        ("Beth", "Bulgaria", 7),
+        ("Carl", "Canada", 30),
+    ]
+    expected_cols = ["name", "country", "population"]
+    expected = DataFrame.from_records(expected_data, columns=expected_cols)
+
+    result = pop.merge(ppl, on=("name", "country"), how="right")
+
+    assert_frame_equal(expected, result)
+
+
+def test_left_merge_preserves_row_order():
```

Comment:
Is this not already accounted for in other tests?

Reply:
None that I could find.

Reply:
Instead of having these as separate tests can you just parametrize on how = left and right?

Reply:
can you do this

Reply:
also outer and inner should match left ordering

Reply:
Sorry - yeah I can do this. Just started a new job so my schedule got crammed all of a sudden.